decidim-core/lib/decidim/migrators/username_to_nickname.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Migrators
+    class UsernameToNickname
+      class User < ApplicationRecord
+        include Decidim::Nicknamizable
+
+        self.table_name = :decidim_users
+      end
+
+      def migrate!
+        User.find_each do |user|
+          user.update!(nickname: User.nicknamize(user.name))
```

Comment:
I added this "migrator" class so I could properly unit test it, but in the end most of the logic went down to the model, so this could be removed and the tests moved around.

Reply:
I ended up removing it because I found a bug in nickname disambiguation and it was hard to test it against the migration. Now I'm testing the `Nicknamizable` mixin directly.